### PR TITLE
Send metrics sooner and add backupStartTime metric

### PR DESCRIPTION
**Pull request by mcanevet**

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func NewMetrics(c *handler.Conplicity) *PrometheusMetrics {
 }
 
 // Push sends metrics to a Prometheus push gateway
-func (p *PrometheusMetrics) Push() (err error) {
+func (p *PrometheusMetrics) Push(method string) (err error) {
```
Comment (raphink) [inline metrics/metrics.go]: I'd rather use a boolean in `Push`, something like

```go
func (p *PrometheusMetrics) Push(replace bool) (err error) {
  var verb string
  if replace {
    verb = "PUT"
  } else {
    verb = "POST"
  }
} 
```

File: conplicity.go [hunk 2]
```diff
@@ -27,23 +28,38 @@ func main() {
 	vols, err := c.GetVolumes()
 	util.CheckErr(err, "Failed to get Docker volumes: %v", "fatal")
 
+	m := metrics.NewMetrics(c)
+
 	for _, vol := range vols {
+		startTimeMetric := fmt.Sprintf("%v{volume=\"%v\",what=\"backupStartTime\"} %v", vol.Name, vol.Name, time.Now().Unix())
```
Comment (raphink) [inline conplicity.go]: I think at that point I'd like to start modelling metrics as a struct:

```go
type Metric struct {
  Volume *volume.Volume
  What string
  Value string
}
```

**Comment by mcanevet** [comment]
I'm not sure about the structure of the metrics we send to prometheus...
Up to 0.21.0 at least we are using job=conplicity, instance=hostname and metric name=conplicity.
This patch will change the metric name to conplicity_<volume_name> so that we can group metric per volume and send result ASAP, but I'm still not sure it's the right way to do it.

**Comment by mcanevet** [comment]
I updated this PR so that it returns this metrics:
- conplicity_backupExitCode
- conplicity_backupStartTime
- conplicity_lastBackup
- conplicity_lastFulleBackup
- conplicity_verifyExitCode

**Comment by mcanevet** [comment]
I merged the metrics renaming (3b66c2ac840707dc620a65dbe9d15c87ff0d2be2)

**Pull request closed without merging**